autoinstrumentation/java/version.txt

```diff
@@ -1 +1 @@
-1.22.1
+1.23.0
```

Comment:
Do we need a changelog entry? @open-telemetry/operator-ta-maintainers

Reply:
I don't think so, unless this is a breaking version upgrade.

Reply:
I think we should include a changelog entry even if it's not breaking as there may be new features added.  Also, we should update to 1.23.1 as there was an issue with 1.23.0.  https://github.com/open-telemetry/opentelemetry-java/releases/tag/v1.23.1

Reply:
Yes, there should be a entry as we change the current expected behavior.

Reply:
Seems a jar for 1.23.1 is missing.

https://github.com/open-telemetry/opentelemetry-operator/actions/runs/4198078109/jobs/7281231001#step:9:175

Reply:
There isn't a Javaagent 1.23.1 release. The Javaagent 1.23.0 release was built on top of Java SDK 1.23.1.

Reply:
That's incredibly confusing, though I can understand why.

Reply:
The changelog is not needed for this PR. It only publishes new image. 


The image is then specified in the operator in the release PR.

Reply:
See https://github.com/open-telemetry/opentelemetry-operator/pull/1484